src/emulator/net/include/net/state.h

```diff
@@ -0,0 +1,55 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <vector>
+
+#include <net/functions.h>
+
+#include <psp2/types.h>
+
+#ifdef _MSC_VER
+# define NOMINMAX
+# define WIN32_LEAN_AND_MEAN
+# define _WINSOCK_DEPRECATED_NO_WARNINGS
```

Comment:
These defines can be put in the CMakeLists.txt that sets up the Winsock interface library. I do something similar for microprofile, to make sure the settings are consistent whenever it's used.